AK/InsertionSort.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022, Marc Luqué <marc.luque@outlook.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Concepts.h>
+#include <AK/StdLibExtras.h>
+
+namespace AK {
+
+// Standard Insertion Sort, with `end` inclusive!
+template<typename Collection, typename Comparator, typename T = decltype(declval<Collection>()[declval<int>()])>
+void insertion_sort(Collection& col, ssize_t start, ssize_t end, Comparator comparator) requires(Indexable<Collection, T>)
+{
+    for (ssize_t i = start + 1; i <= end; ++i) {
```

Comment:
Is the topmost loop parentesis redundant?

Reply:
We don't alleviate parenthesis when the following contains multiple lines.